### PR TITLE
server: allow cors http request

### PR DESCRIPTION
**Pull request by Kingwl**
### What problem does this PR solve? 
https://github.com/pingcap/tidb/issues/7850

### What is changed and how it works?
add a cors middleware and config

### Check List 

Tests 

 - No code

Related changes

 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Comment by shenli** [comment]
@Kingwl Thanks!

**Comment by shenli** [comment]
Please fix the CI.

**Comment by Kingwl** [comment]
wait a few moment

**Comment by zz-jason** [comment]
@jackysp @lysu PTAL

File: server/util.go [hunk 1]
```diff
@@ -354,3 +356,17 @@ func appendFormatFloat(in []byte, fVal float64, prec, bitSize int) []byte {
 	}
 	return out
 }
+
+// CorsHandler add Cors Header if `cors` config is setted
```
Comment (XuHuaiyu) [inline server/util.go]: s/ setted/ set
s/ add/ adds
add a `.` at the end of this comment

File: server/util.go [hunk 2]
```diff
@@ -37,8 +37,10 @@ package server
 import (
 	"bytes"
 	"encoding/binary"
+	"github.com/pingcap/tidb/config"
```
Comment (XuHuaiyu) [inline server/util.go]: put this import below the native library, and leave an empty line between them.

File: config/config.go [hunk 3]
```diff
@@ -252,6 +253,7 @@ var defaultConf = Config{
 	Host:             "0.0.0.0",
 	AdvertiseAddress: "",
 	Port:             4000,
+	Cors:             false,
```
Comment (lysu) [inline config/config.go]: @xuechunL  How do you want to use those api~?

default disable `cors` and enable when you app deploy with tidb is ok~?

File: server/util.go [hunk 4]
```diff
@@ -354,3 +356,17 @@ func appendFormatFloat(in []byte, fVal float64, prec, bitSize int) []byte {
...
+
+// CorsHandler add Cors Header if `cors` config is setted
+type CorsHandler struct {
+	handler http.Handler
+	cfg     *config.Config
+}
+
+func (h CorsHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	if h.cfg.Cors {
+		w.Header().Set("Access-Control-Allow-Origin", "*")
```
Comment (lysu) [inline server/util.go]: @xuechunL do we need set more special address?

if so maybe we can change configuration from enable/disable to a host string? and set that value via ansible?

**Pull request merged**